hawkular/client.py

```diff
@@ -189,7 +189,10 @@ def _http(self, url, method, data=None, decoder=None):
             res = urlopen(req, context=self.context)
 
             if res.getcode() == 200:
-                data = json.load(reader(res), cls=decoder)
+                try:
+                    data = json.load(reader(res), cls=decoder)
+                except ValueError:
+                    data = reader(res).read()
```

Comment:
@cben I'll only answer to the JSON question now as I haven't had time to review rest of the thing. I would like to encapsulate the fact that the transfer method is JSON at this current point as we have plans to support other transportation protocols also in the future (for some use-cases at least). 

However, I'm not against allowing user to supply their own parsing method/function or to get JSON without using automatic parsing. It could be possible to supply some generic function for everyone and then allow "advanced use-cases" to overwrite that one. That's somewhat the idea I've pushed towards in the Golang client. Would that sort of approach work on what you had in mind?